src/scanner.mll

```diff
@@ -49,15 +49,14 @@ let string = '"' ascii* '"'
 rule token = parse
 ['\t' '\r'] { token lexbuf } (* whitespace *)
 | ' '       { if !is_start_of_line then new_spacing := !new_spacing + 1 else (); token lexbuf }
-| '\n'      { set_new_line (); token lexbuf } 
-| '#'       { is_start_of_line := false; comment lexbuf } (* comment *)
+| '\n'      { let was_start = !is_start_of_line in set_new_line (); if not was_start then EOL::(token lexbuf) else token lexbuf } 
```

Comment:
differentiate between new line at end of a statement (use EOL token) and empty new line (ignore)

Reply:
genius